### ISSUE TITLE
Problem with the pre-release version suffix, which ends with a number greater than 9

### ISSUE THREAD

**Issue by Taritsyn** [NuGet/Home#6383]
Having released a [package with version `3.0.0-alpha10`](https://www.nuget.org/packages/JavaScriptEngineSwitcher.V8/3.0.0-alpha10), I faced into the problem of updating projects to this version via Visual Studio UI (NuGet Package Manager for Visual Studio 2013 version 2.12.0.817 and Visual Studio 2017 version 15.5.2 with NuGet Package Manager version 4.5.0). Problem is that the pre-release version suffix is processed as an whole string and this leads to wrong ordering of versions:

```
3.0.0-alpha9
3.0.0-alpha7
3.0.0-alpha6
3.0.0-alpha5
3.0.0-alpha2
3.0.0-alpha10
3.0.0-alpha1
```

In this case, version `3.0.0-alpha10` is not considered to be the latest version, but version `3.0.0-alpha9` is still considered the latest version.

I understand that my problem is very specific and most developers will not face it. If you can not fix it, then at least describe this problem in documentation.

**Comment by zhili1208** [comment]
Please use Semver 2.0.0, like "3.0.0-alpha.10" https://semver.org/

**Comment by Taritsyn** [comment]
@zhili1208 Thanks for information! But even if use this notation (`3.0.0-alpha.10`) in Visual Studio 2017, then still version `3.0.0-alpha9` will be considered the last.

In addition, this notation is not supported in Visual Studio 2013 and 2015. In Visual Studio 2015, using this notation causes an error:

```
'3.0.0-alpha.10' is not a valid version string.
Parameter name: version
```

**Comment by dasMulli** [comment]
The label `alpha9` is still considered higher than `alpha`, so if you had been using SemVer 2 all along, it would choose `-alpha.10` over `-alpha.9` but not over `-alpha9`.

This is why many counter or commit count based versions that are backwards compatible use some form of padding, e.g. `-alpha-0009`, `alpha-0010` etc.

**Comment by Taritsyn** [comment]
> The label alpha9 is still considered higher than alpha, so if you had been using SemVer 2 all along, it would choose -alpha.10 over -alpha.9 but not over -alpha9.

I already understood that myself. Can you give an example of a package on www.nuget.org, that supports a new notation?

> This is why many counter or commit count based versions that are backwards compatible use some form of padding, e.g. -alpha-0009, alpha-0010 etc.

I think that for me it's too late to use this approach.

**Comment by nkolev92** [comment]
@Taritsyn 
Unfortunately, at this point, there isn't much you can do other than drastically changing you prerelease versions. 

Semver 2.0 is available for VS 2015, however, as we shipped a downlevel version 3.6.0, which you can find on nuget.org/downloads. 
Unfortunately the warning on nuget.org is outdated, so I have filled an issue to get it updated. https://github.com/NuGet/NuGetGallery/issues/5244

So your 2 options here are once you switch to a newer label:
- Use @dasMulli's recommendation with padding
- Switch to semver 2.0, with the risk of making your package incompatible for VS 2013 users and earlier. 

There's some Semver 2.0 information at the following links:

https://blog.nuget.org/20170815/Whats-nu-in-NuGet-with-VS2017-15-3.html
Search for "Visual Studio 2015" on the page. 

https://github.com/NuGet/Home/wiki/SemVer-2.0.0-support

**Comment by nkolev92** [comment]
Closing this now. 
Happy to answer any further questions.
